mgnify_pipelines_toolkit/analysis/shared/amrintegrator.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright 2024-2025 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import argparse
+import gzip
+import os
+
+
+def validate_inputs(optional_inputs):
+    valid_inputs = []
+    for name, path in optional_inputs.items():
+        if path and os.path.exists(path):
+            valid_inputs.append(name)
+        elif path:
+            print(f"Warning: file not found for '{name}' → {path}")
+    return valid_inputs
+
+
+def open_file(filename):
+    """
+    Open a file, handling both compressed (.gz) and uncompressed files.
+    Returns a file handle that can be used for reading.
+    """
+    if filename.endswith(".gz"):
+        return gzip.open(filename, "rt")  # 'rt' for text mode
+    else:
+        return open(filename, "r")
+
+
+def parse_hamronized(amr_annotation, input_file):
```

Comment:
I think `hamronized` is a typo? :D if so:
```suggestion
def parse_harmonized(amr_annotation, input_file):
```